tests/unit/Entity/EntityIdValueTest.php

```diff
@@ -75,41 +78,95 @@ public function testGetSortKey( EntityIdValue $id ) {
 		$this->assertInternalType( 'string', $id->getSortKey() );
 	}
 
+	public function provideGetArrayValue() {
+		return [
+			'Q2147483647' => [
+				new ItemId( 'Q2147483647' ),
+				[
+					'entity-type' => 'item',
+					'numeric-id' => 2147483647,
+					'id' => 'Q2147483647'
+				],
+			],
+			'P31337' => [
+				new PropertyId( 'P31337' ),
+				[
+					'entity-type' => 'property',
+					'numeric-id' => 31337,
+					'id' => 'P31337',
+				],
+			],
+			'X567' => [
+				$this->newCustomId( 'X567' ),
+				[
+					'entity-type' => 'custom',
+					'id' => 'X567',
+				],
+			],
+			'foo:P678' => [
+				new PropertyId( 'foo:P678' ),
+				[
+					'entity-type' => 'property',
+					'numeric-id' => 678,
+					'id' => 'foo:P678',
+				],
+			],
+		];
+	}
+
 	/**
-	 * @dataProvider instanceProvider
+	 * @dataProvider provideGetArrayValue
 	 */
-	public function testGetArrayValueRoundtrip( EntityIdValue $id ) {
-		$newId = EntityIdValue::newFromArray( $id->getArrayValue() );
+	public function testGetArrayValue( EntityId $id, array $expected ) {
+		$value = new EntityIdValue( $id );
+		$array = $value->getArrayValue();
 
-		$this->assertEquals( $id, $newId );
+		ksort( $expected );
+		ksort( $array );
+		$this->assertEquals( $expected, $array );
```

Comment:
We are again talking about two entirely different idea of what "data model" means.

This is a PHP implementation that encodes a major part, but not al knowledge about the conceptual data model. Giving a guarantee about the order of array keys in a serialization is convenient and makes code that depends on this simpler and easier to maintain. It does not mean all code consuming this array must keep the key order, especially not if it just can't, as it is the case in JSON.